android/app/src/main/java/com/google/samples/cronet_sample/MainActivity.java

```diff
@@ -123,4 +138,49 @@ public void run() {
             this.cronetLatency.set(averageLatency);
         }
     }
+
+    private synchronized void setCronetEngine() {
```

Comment:
Instead of custom locking and initialization, can you use the static inner class holder pattern? E.g. https://stackoverflow.com/a/71683

Reply:
I removed the syncing totally because I just made croentEngine private with a getter and it's only initialized in the onCreate().